### ISSUE TITLE
Calendar invite handling is not supported in Thunderbird 115.x

### ISSUE THREAD

**Issue by Standard8** [thunderbird-conversations/thunderbird-conversations#2007]
Due to the impact of the changes from Thunderbird 102 to Thunderbird 115 (all of which have been good improvements for the future of Thunderbird), Conversations has required significant parts rewriting to continue working. Unfortunately due to lack of time, not all of it currently supports 115. Given that 115 is now out, it was considered better to release something that mainly works, and re-add support for the remaining parts as time becomes available.

This issue is for the calendar notification bar that gets displayed when an invite is detected.

The work here is to go through Thunderbird's UI and extract the relevant parts necessary to provide the appropriate calendar handling, and figure out how to fit those into Conversations' architecture

The workaround is to view the message in the classic reader.

**Comment by MLopez-Ibanez** [comment]
Is there any TB bug or improvement that could help with this? It is not great that Thunderbird seems to be moving towards having conversations but not getting there, yet the conversations add-on seems to break more with every TB release rather than less. The transition to 102 did not produce so much breakage.

**Comment by Standard8** [comment]
> Is there any TB bug or improvement that could help with this? It is not great that Thunderbird seems to be moving towards having conversations but not getting there, yet the conversations add-on seems to break more with every TB release rather than less. The transition to 102 did not produce so much breakage.

TB 115 was a much bigger shift than TB 102. TB 102 didn't really change the UI much.

Prior to 115, the tabs weren't really "tabs". It was one three-pane view that was re-used across the tabs and adjusted to display the appropriate item as you switched.

After 115, tabs are now separate entities, each one is a three-pane view (or message view) independently of the others. Additionally, the parts responsible for the message display (including calendar & encrypted email handling) are now loaded separately in their own browser element rather than being loaded alongside the main three pane view and are managed slightly differently.

All this meant a big shift for Conversations as well. Now Conversations has to take account of the truly independent tabs. That has been done, and as a result some of the management there has been shifted to make more use of the Thunderbird provided WebExtension APIs rather than custom hooks we had previously. For future maintainability that is a big improvement, because it is using the provided APIs, but it has also simplified a lot of the code - especially the custom code that we use to hook Conversations into the correct places.

Unfortunately, the change for the message display means that Conversations can't hook into some of the functions it did before - they might not even be loaded if you don't display a message in the classic reader. This is what is driving the change here and for the encrypted issue.

Whilst it might be possible to load those missing functions, unfortunately they are integrated deeply into Thunderbird and expect Thunderbird's UI, that it really means that rewriting them for Conversations is the best option.

So overall, the shift has been a good one. Thunderbird has fixed some long standing issues with the shift to truly independent tabs, which has also helped Conversations to reduce complexity and shift to more WebExtension APIs.

There's obviously been a lot of work to do, and hopefully I'll be able to get a bit of time to finish it off soon.

**Comment by Standard8** [comment]
I have just finished re-implementing the support for Calendar invites 🎉 .

The implementation should work pretty much the same as Thunderbird's native implementation, as it is mainly re-using code from that.

I've just pushed up a new version (4.1.4) onto the add-on site, and it is now awaiting review. It is normally a few days before it gets review, but hopefully it won't be too long now.

Although we could still do with a patch for Thunderbird to simplify what Conversations needs to implement, I've copied the necessary code into the add-on for now, so that this could be re-enabled sooner.
